### ISSUE TITLE
segfaults when calling setup

### ISSUE THREAD

**Issue by MurdeRM3L0DY** [noib3/nvim-completion#19]
Hi. Would you mind sharing what versions of neovim, nvim-oxi and rustc you use to test nvim-completion?
I've tried with rustc 1.65.0-nightly/1.63.0, neovim (NVIM v0.8.0-dev-1021-g24fbda04b
Build type: Release), nvim-oxi latest master. Could you also share your config options for nvim-completion? Thanks.

**Comment by noib3** [comment]
- rustc: `rustc 1.65.0-nightly (e1b28cd2f 2022-08-19)`;
- nvim:  `NVIM v0.7.2`;
- nvim-oxi: latest master, updated in (d6f48b06a448b73135dafbfbc771ce9dd61efead);

Config:

```lua
-- config.lua

local completion = require("nvim_completion")

completion.setup({
  sources = {
    lipsum = { enable = function(_buf) return true end },
    lsp = { enable = true },
  },
})
```

Starting Neovim w/ `nvim --clean -u ./config.lua`.

Then you need to open a new buffer for the sources to attach to via `:e <somefilename>`.

Can you share your config? It shouldn't segfault. Does the segfault also happen if you use Neovim 0.7.2 instead of nightly?

**Comment by MurdeRM3L0DY** [comment]
Yup, just confirmed it only segfaults with neovim latest master.

**Comment by noib3** [comment]
That's not a segfault, that's just a panic (which I'm aware of).

**Comment by MurdeRM3L0DY** [comment]
Yup, I figured you would.
